classes/PodsAPI.php

```diff
@@ -2885,42 +2885,106 @@ public function save_field( $params, $table_operation = true, $sanitized = false
+				 * @param string             $type            The field type.
+				 * @param Pods\Whatsit\Field $field           The field object.
+				 * @param array              $extra_info      {
+				 *      Extra information about the field.
+				 *
+				 *      @type bool        $simple Whether the field is a simple tableless field.
+				 *      @type string      $definition The field definition.
+				 *      @type null|string $old_name The old field name (if preexisting).
+				 *      @type null|string $old_definition The old field definition (if preexisting).
+				 *      @type null|array  $old_options The old field options (if preexisting).
+				 * }
+				 */
+				do_action( 'pods_api_save_field_table_pre_alter', $definition_mode, $pod, $field['type'], $field, [
+					'simple'         => $simple,
+					'definition'     => $definition,
+					'old_name'       => $old_name,
+					'old_definition' => $old_definition,
+					'old_options'    => $old_options,
+				] );
+
+				if ( 'drop' === $definition_mode ) {
+					// Drop field column.
+					pods_query( "ALTER TABLE `@wp_pods_{$params->pod}` DROP COLUMN `{$old_name}`", false );
+				} elseif ( 'change' === $definition_mode ) {
+					// Change field column definition.
+					if ( $old_name && $old_name !== $field['name'] ) {
+						$test = pods_query( "ALTER TABLE `@wp_pods_{$params->pod}` CHANGE `{$old_name}` {$definition}", false );
+					} else {
+						$test = pods_query( "ALTER TABLE `@wp_pods_{$params->pod}` MODIFY {$definition}", false );
+					}
+
+					if ( false === $test ) {
+						$definition_mode = 'add';
+					}
 				}
 
 				// If the old field doesn't exist, continue to add a new field
-				if ( false === $test ) {
+				if ( 'add' === $definition_mode ) {
 					pods_query( "ALTER TABLE `@wp_pods_{$params->pod}` ADD COLUMN {$definition}", __( 'Cannot create new field', 'pods' ) );
 				}
+
+				/**
+				 * Allow hooking into after the table has been altered for any custom functionality needed.
+				 *
+				 * @since TBD
+				 *
+				 * @param string             $definition_mode The definition mode used for the table field.
+				 * @param Pods\Whatsit\Pod   $pod             The pod object.
+				 * @param string             $type            The field type.
+				 * @param Pods\Whatsit\Field $field           The field object.
+				 * @param array              $extra_info      {
+				 *      Extra information about the field.
+				 *
+				 *      @type bool        $simple Whether the field is a simple tableless field.
+				 *      @type string      $definition The field definition.
+				 *      @type null|string $old_name The old field name (if preexisting).
+				 *      @type null|string $old_definition The old field definition (if preexisting).
+				 *      @type null|array  $old_options The old field options (if preexisting).
+				 * }
+				 */
+				do_action( 'pods_api_save_field_table_altered', $definition_mode, $pod, $field['type'], $field, [
```

Comment:
* :no_entry_sign: Opening parenthesis of a multi-line function call must be the last content on the line
* :no_entry_sign: Only one argument is allowed per line in a multi-line function call
* :no_entry_sign: Only one argument is allowed per line in a multi-line function call
* :no_entry_sign: Only one argument is allowed per line in a multi-line function call
* :no_entry_sign: Only one argument is allowed per line in a multi-line function call
* :no_entry_sign: Only one argument is allowed per line in a multi-line function call
* :no_entry_sign: Short array syntax is not allowed

<sub>hash: 995205b133016080601edced55065a34</sub>